### PR TITLE
DDF-3469 Sanitize passwords from the security log

### PR DESCRIPTION
**Pull request by blen-desta**
#### What does this PR do?
Auditing the action of a configuration (created, updated ...) with the pid instead of the entire configuration.

#### Who is reviewing it? 
@brjeter @tbatie 

#### Choose 2 committers to review/merge the PR. 
@coyotesqrl
@rzwiefel

#### How should this be tested? (List steps with links to updated documentation)
Verify that passwords are not printed in plain text in the security log by
* Creating and updating sources from the Admin Console and the Sources Wizard
* Connecting to LDAP through the Admin Console and the LDAP Wizard
   - Login and LDAP Claims Handler from the Admin Console 
   - Authentication and Attribute store from the LDAP Wizard

#### What are the relevant tickets?
[DDF-3469](https://codice.atlassian.net/browse/DDF-3469)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
For the general case, merging is gated on the following:

* Review by at least two contributors
* Review by at least two committers
* Successful CI
* Successful static analysis (SonarQube/Coverity/etc.)
* Heroing of change (details vary by ticket)

Certain exceptions may occur. For example, if there is a problem with the CI environment then a manually performed full build and static analysis check may be substituted _at the discretion of the repository maintainers_. Additionally, repository metadata such as this `PULL_REQUEST_TEMPLATE`, the repository `README`, and other non-functional items may be merged without requiring CI and static analysis.

Finally, an abbreviated review process may be in place for forward-port tickets immediately following branching and prior to branch divergence. In that case, approvals, CI, static analysis, and heroing that occur against the PR against the _release branch_ can be considered sufficient for reviewing the PR against the _master branch_.

Once the branches diverge, the process will return to normal.

Insofar as divergence is not a function of time but of change to the parent branch, the repository maintainers will determine when abbreviated review ends.

When abbreviated review is in effect, a note to that effect will be at the top of this template. Following is an example note:
```
##### ABBREVIATED REVIEW BETWEEN 2.12.X AND MASTER IS IN EFFECT
____

```

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/security/core/security-core-impl/src/main/java/ddf/security/config/impl/ConfigurationSecurityLogger.java [hunk 1]
```diff
@@ -75,8 +83,17 @@ public void configurationEvent(ConfigurationEvent event) {
   }
 
   @SuppressWarnings("squid:S00112" /* Generic RuntimeException is acceptable here */)
-  private String dictionaryToString(Dictionary<String, Object> properties) {
+  @VisibleForTesting
+  String dictionaryToString(Dictionary<String, Object> properties) {
     ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+    if (properties.remove(PASSWORD) != null) {
```
Comment (paouelle) [inline platform/security/core/security-core-impl/src/main/java/ddf/security/config/impl/ConfigurationSecurityLogger.java]: ❓ How do we know these are the only 2 password properties that should be sanitized? I though this info was stored in the meta types.

**Comment by blen-desta** [comment]
build now

**Comment by Lambeaux** [comment]
Hero'd 👍

**Comment by blen-desta** [comment]
Pushed changes
build now

**Comment by blen-desta** [comment]
Build failed because of an OWASP vulnerability with proxy-camel-servlet-2.12.0-SNAPSHOT.jar (CVE-2017-12634, CVE-2017-12633)
Building again because the fix was merged
Build now

**Comment by blen-desta** [comment]
Unrelated failure, building again
Build now

**Comment by blen-desta** [comment]
Build failed because `TestCatalogSearchUi` failed to setup.
Issue not related to PR changes.
build now

**Comment by blen-desta** [comment]
Build failed with:
`Exception during gitflow-incremental-builder execution occurred. Git branch of name 'refs/remotes/origin/master' not found`, building again
Build now

**Pull request merged**